src/react/RegisterDevicePage.js

```diff
@@ -203,9 +278,64 @@ export default class RegisterDevicePage extends React.Component {
     this.setNewDeviceId(newDeviceID);
   }
 
-  handleModalCancel() {
+  credentialModalCancel() {
+    this.setState({
+      showCredentialSecretModal: false
+    });
+  }
+
+  onNamespaceChange(e) {
+    const namespace = e.target.value;
+
+    this.setState({
+      namespace: namespace
+    });
+    this.maybeGenerateDeviceId(namespace, this.state.customString);
+  }
+
+  onCustromStringChange(e) {
+    const customString = e.target.value;
+
+    this.setState({
+      customString: customString
+    });
+    this.maybeGenerateDeviceId(this.state.namespace, customString);
+  }
+
+  showNamespaceModal() {
+    this.setState({
+      showNamespaceModal: true
+    });
+  }
+
+  namespaceModalCancel() {
+    this.setState({
+      showNamespaceModal: false
+    });
+  }
+
+  maybeGenerateDeviceId(namespace, customString) {
+    let newDeviceID = "";
+
+    try {
+      const newUUID = uuidv5(customString, namespace).replace(/-/g, "");
+      const bytes = newUUID.match(/.{2}/g).map(b => parseInt(b, 16));
+      newDeviceID = byteArrayToUrlSafeBase64(bytes);
+    } catch (e) {
+      // namespace is not a UUID
```

Comment:
are you sure about this empty catch block?

Reply:
i can make this more explicit, but yes, It is intended.

Reply:
I'll rewrite this to make it clearer